paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java

```diff
@@ -97,18 +100,31 @@ public class HiveCatalog extends AbstractCatalog {
     private final String clientClassName;
     private final IMetaStoreClient client;
 
+    private LocationHelper locationHelper;
+
     public HiveCatalog(FileIO fileIO, HiveConf hiveConf, String clientClassName) {
-        super(fileIO);
-        this.hiveConf = hiveConf;
-        this.clientClassName = clientClassName;
-        this.client = createClient(hiveConf, clientClassName);
+        this(fileIO, hiveConf, clientClassName, Collections.emptyMap());
     }
 
     public HiveCatalog(
             FileIO fileIO, HiveConf hiveConf, String clientClassName, Map<String, String> options) {
         super(fileIO, options);
         this.hiveConf = hiveConf;
         this.clientClassName = clientClassName;
+
+        boolean needLocationInProperties =
+                hiveConf.getBoolean(
+                        LOCATION_IN_PROPERTIES.key(), LOCATION_IN_PROPERTIES.defaultValue());
+        if (needLocationInProperties) {
+            locationHelper = new TBPropertiesLocationHelper(hiveConf);
+        } else {
+            // set the warehouse location to the hiveConf
+            String warehouse = hiveConf.get(CatalogOptions.WAREHOUSE.key());
+            hiveConf.set(HiveConf.ConfVars.METASTOREWAREHOUSE.varname, warehouse);
+
+            locationHelper = new StorageLocationHelper(hiveConf);
```

Comment:
We can put these two lines to the constructor of `StorageLocationHelper`.
```
String warehouse = hiveConf.get(CatalogOptions.WAREHOUSE.key());
hiveConf.set(HiveConf.ConfVars.METASTOREWAREHOUSE.varname, warehouse);
```

Reply:
if we pass warehouse to `HiveCatalog` at the above conversation , we can specify the warehouse directly .So that  we don't need pass `hiveConf` for  specifying the warehouse at the both `LocationHelper` separately . Shall we retain these two lines  outside the `LocationHelper`.